File: python/Crud.py
class Crud:
  __list = [] # deklarasi atribut private list untuk menyimpan objek Mahasiswa
  __listNim = [] # deklarasi atribut private list untuk menyimpan NIM dari objek Mahasiswa

  def tambah(self, mhs):
    # method untuk menambahkan objek Mahasiswa ke dalam list __list dan NIM ke dalam list __listNim
    if(mhs.getNim() in self.__listNim): # cek apakah NIM sudah terdaftar pada list __listNim
      print("data yang ingin anda tambah tidak unique value") # jika sudah, cetak pesan bahwa data tidak unik
      return
    self.__list.append(mhs) # tambahkan objek Mahasiswa ke dalam list __list
    self.__listNim.append(mhs.getNim()) # tambahkan NIM ke dalam list __listNim
    
  def tampil(self):
    # method untuk menampilkan seluruh objek Mahasiswa pada list __list beserta atribut-atributnya
    # print("keren")
    for i in range(len(self.__list)): # loop sebanyak panjang dari list __list
      print( 
      str(i + 1) 
      , "\nNIK : " , self.__list[i].getNik()
      , "\nNIM : " , self.__list[i].getNim()
      , "\nNama : " , self.__list[i].getNama()
      , "\nJenis Kelamin : " , self.__list[i].getJenisKelamin()
      , "\nAsal universitas : " , self.__list[i].getAsalUniv()
      , "\nEmail Edu : " , self.__list[i].getEmailEdu()
      , "\nProdi : " , self.__list[i].getProdi()
      , "\nFakultas : " , self.__list[i].getFakultas()
      , "\n"
      )
      # print()

File: c++/Crud.cpp
class Crud{
  private:
    vector<Mahasiswa> list;     // atribut list dengan tipe data vector of class Mahasiswa
    vector<string> listNim;     // atribut list NIM dengan tipe data vector of string
  public:
    // Konstruktor
    Crud(){}

    // Method untuk menambahkan objek Mahasiswa ke dalam list
    void tambah(Mahasiswa mhs){
      // Cek apakah NIM mahasiswa yang ingin ditambahkan sudah ada di dalam list
      if(find(listNim.begin(), listNim.end(), mhs.getNim()) != listNim.end()){
        cout << "Data yang ingin Anda tambah tidak unique value" << endl;
        return;
      }
      // Jika NIM belum ada di dalam list, tambahkan Mahasiswa ke dalam list dan tambahkan NIM ke dalam listNim
      this->list.push_back(mhs);
      this->listNim.push_back(mhs.getNim());
    }

    // Method untuk menampilkan semua objek Mahasiswa dalam list
    void tampil(){
      for(int i = 0; i < list.size(); i++){
        cout << 
        (i + 1) 
        << "\nNIK : " << list[i].getNik()
        << "\nNIM : " << list[i].getNim()
        << "\nNama : " << list[i].getNama()
        << "\nJenis Kelamin : " << list[i].getJenisKelamin()
        << "\nAsal universitas : " << list[i].getAsalUniv()
        << "\nEmail Edu : " << list[i].getEmailEdu()
        << "\nProdi : " << list[i].getProdi()
        << "\nFakultas : " << list[i].getFakultas()
        << "\n";
      }
    }
};